pgml-sdks/python/pgml/pgml/collection.py

```diff
@@ -0,0 +1,771 @@
+            % (self.splitters_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS name_index ON %s (name);"
+            % (self.splitters_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS parameters_index ON %s USING GIN (parameters jsonb_path_ops);"
+            % (self.splitters_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        self.pool.putconn(conn)
+
+    def _create_models_table(self) -> None:
+        """
+        This function creates a table in a PostgreSQL database with specific columns and indexes.
+        """
+        conn = self.pool.getconn()
+        self.models_table = self.name + ".models"
+        create_statement = (
+            "CREATE TABLE IF NOT EXISTS %s (\
+                            id          serial8 PRIMARY KEY, \
+                            created_at  timestamptz NOT NULL DEFAULT now(), \
+                            task        text NOT NULL, \
+                            name        text NOT NULL, \
+                            parameters  jsonb NOT NULL DEFAULT '{}'\
+                    );"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, create_statement)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS created_at_index ON %s (created_at);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS task_index ON %s (task);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS name_index ON %s (name);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS parameters_index ON %s USING GIN (parameters jsonb_path_ops);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        self.pool.putconn(conn)
+
+    def _create_transforms_table(self) -> None:
+        """
+        This function creates a transforms table in a PostgreSQL database and adds indexes to it.
+        """
+        conn = self.pool.getconn()
+        self.transforms_table = self.name + ".transforms"
+        create_statement = (
+            "CREATE TABLE IF NOT EXISTS %s (\
+                            oid         regclass PRIMARY KEY,\
```

Comment:
Odd way of referencing another table, I've never seen that done before. Is this compatible with logical replication? I.e. if we want to move this data to another database, the oids probably won't match anymore, since they are specific to a Postgres installation.

Reply:
There is some documentation about this but OID is not the same as the designated registered types -- `regclass`, `regrole` and so on.

Reply:
This should probably be called `table_name`. An `oid` is something very Postgres-specific and does not mean a table reference at all. In fact,it can reference a row in a TOAST table or a type in `pg_type`.

Reply:
It could be called `table_name`, `data_table`, `tab`, &c -- there is nothing wrong with that.

An `oid` is any object ID -- in `pg_class`, the ID of the table is in fact called `oid`. An `oid` the refers to an actual relation can be casted to `regclass`, otherwise not. Since we're dynamically adding and removing tables, it pays to  be conversant with how the catalog describes these things.